### PR TITLE
Fix bug and edit doctests for infix_to_prefix_conversion

### PR DESCRIPTION
**Pull request by hollowcrust**
### Describe your change:

Fixes #10063 and #10095.

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: data_structures/stacks/infix_to_prefix_conversion.py [hunk 1]
```diff
@@ -83,7 +100,11 @@ def infix_2_postfix(infix):
             if len(stack) == 0:
                 stack.append(x)  # If stack is empty, push x to stack
             else:  # while priority of x is not > priority of element in the stack
-                while len(stack) > 0 and priority[x] <= priority[stack[-1]]:
+                while (
+                    len(stack) > 0
```
Comment (cclauss) [inline data_structures/stacks/infix_to_prefix_conversion.py:104]: This is slightly safer if `stack` is set to `None`
```suggestion
                    stack
```

**Comment by cclauss** [comment]
Did the previous pull request have other enhancements?

File: data_structures/stacks/infix_to_prefix_conversion.py [hunk 2]
```diff
@@ -134,24 +166,20 @@ def infix_2_prefix(infix):
...
-       )     | (       |
-             |         | (
-    '('
+    >>> infix_2_prefix('())')
+    Traceback (most recent call last):
+        ...
+    ValueError: invalid expression
     """
-    infix = list(infix[::-1])  # reverse the infix equation
+    new_infix = list(infix[::-1])  # reverse the infix equation
```
Comment (cclauss) [inline data_structures/stacks/infix_to_prefix_conversion.py:174]: Would `reversed_infix` be a more self-documenting name to help the reader understand?

**Comment by hollowcrust** [comment]
Yes, I just brought the enhancements from the old PR.

File: data_structures/stacks/infix_to_prefix_conversion.py (hunk 2 above)
Comment (hollowcrust) [reply data_structures/stacks/infix_to_prefix_conversion.py:174]: Hmm yea that makes more sense

File: data_structures/stacks/infix_to_prefix_conversion.py [hunk 3]
```diff
@@ -134,24 +166,20 @@ def infix_2_prefix(infix):
...
+    for i in range(len(new_infix)):
+        if new_infix[i] == "(":
+            new_infix[i] = ")"  # change "(" to ")"
+        elif new_infix[i] == ")":
+            new_infix[i] = "("  # change ")" to "("
 
-    return (infix_2_postfix("".join(infix)))[
+    return (infix_2_postfix("".join(new_infix)))[
         ::-1
     ]  # call infix_2_postfix on Infix, return reverse of Postfix
```
Comment (cclauss) [inline data_structures/stacks/infix_to_prefix_conversion.py:184]: Let's not wrap lines of code just to fit in a comment.
```suggestion
    # call infix_2_postfix on Infix, return reverse of Postfix
    return (infix_2_postfix("".join(new_infix)))[::-1]
```

**Review by cclauss — APPROVED** [review]
Awesome work!  Thanks massively.

**Comment by hollowcrust** [comment]
Thanks a lot for your help as well!

**Pull request merged**